MSAL/test/integration/native_auth/end_to_end/MSALNativeAuthEndToEndBaseTestCase.swift

```diff
@@ -26,54 +26,65 @@ import XCTest
-
-    class Configuration: NSObject {
-        static let clientId = ProcessInfo.processInfo.environment["clientId"] ?? "<clientId not set>"
-        static let authorityURLString = ProcessInfo.processInfo.environment["authorityURL"] ?? "<authorityURL not set>"
-    }
-
-    func mockResponse(_ response: MockAPIResponse, endpoint: MockAPIEndpoint) async throws {
-        try await mockAPIHandler.addResponse(
-            endpoint: endpoint,
-            correlationId: correlationId,
-            responses: [response]
-        )
+    class Constants {
+        static let nativeAuthKey = "native_auth"
+        static let clientIdEmailPasswordKey = "email_password_client_id"
+        static let clientIdEmailCodeKey = "email_code_client_id"
+        static let clientIdEmailPasswordAttributesKey = "email_password_attributes_client_id"
+        static let clientIdEmailCodeAttributesKey = "email_code_attributes_client_id"
+        static let tenantSubdomainKey = "tenant_subdomain"
+        static let signInEmailPasswordUsernameKey = "sign_in_email_password_username"
+        static let signInEmailCodeUsernameKey = "sign_in_email_code_username"
+        static let resetPasswordUsernameKey = "reset_password_username"
     }
+    
+    let correlationId = UUID()
+    let defaultTimeout: TimeInterval = 20
 
-    override func tearDown() {
-        try? mockAPIHandler.clearQueues(correlationId: correlationId)
-    }
+    private var confFileContent: [String: String]? = nil
+    private let codeRetriever = MSALNativeAuthEmailCodeRetriever()
 
     override func setUpWithError() throws {
         try super.setUpWithError()
-        let useMockAPIBooleanString = ProcessInfo.processInfo.environment["useMockAPI"] ?? "false"
-        usingMockAPI = Bool(useMockAPIBooleanString) ?? false
         
-        // mock API URL needs to contains a tenant
-        guard let authorityURL = URL(string: Configuration.authorityURLString + (usingMockAPI ? "/testTenant" : "")), let authority = try? MSALCIAMAuthority(url: authorityURL) else {
-            XCTFail("AuthorityURL not set or invalid")
+        guard let confURL = Bundle(for: Self.self).url(forResource: "conf", withExtension: "json"), let configurationData = try? Data(contentsOf: confURL) else {
+            XCTFail("conf.json file not found")
             return
         }
-
-        sut = try MSALNativeAuthPublicClientApplication(
-            configuration: MSALPublicClientApplicationConfig(
-                clientId: Configuration.clientId,
-                redirectUri: nil,
-                authority: authority
-            ),
-            challengeTypes: [.OOB, .password]
-        )
-
-        if usingMockAPI {
-            print("🤖🤖🤖 Using mock API: \(Configuration.authorityURLString)")
-        } else {
-            print("👩‍💻👩‍💻👩‍💻 Using test tenant: \(Configuration.authorityURLString)")
+        let confFile = try? JSONSerialization.jsonObject(with: configurationData, options: []) as? [String: Any]
```

Comment:
I would deserialize the `configurationData` catching the error because it can help us to understand possible errors in the future.

Reply:
Are you suggesting failing the test if the JSONSerialization fails?

Reply:
Yes, it's a rare case, but it is just to fail as soon as possible, rather than waiting for when the confFile is used to find out that is nil because it couldn't be deserialized.

Reply:
Sure, I'll change it